HACKING.md

```diff
@@ -92,6 +92,129 @@ Package build dependencies for other distributions can be found under the
+16.04 as the base for building and Ubuntu 16.04 is in Extended Security 
+Maintenance (ESM), and as such only is buildable using snapcraft's 4.x channel.
+
+Install snapcraft from the 4.x channel:
+
+```
+sudo snap install snapcraft --channel=4.x
+```
+
+Then run snapcraft:
+
+```
+snapcraft
+```
+
+Now the snapd snap that was just built can be installed with:
+
+```
+snap install --dangerous snapd_*.snap
+```
+
+To go back to using snapd from the store instead of the custom version we 
+installed (since it will not get updates as it was installed dangerously), you
+can either use `snap revert snapd`, or you can refresh directly with 
+`snap refresh snapd --stable --amend`.
+
+Note: It is also sometimes useful to use snapcraft to build the snapd snap for
+other architectures using the remote-build feature, however there is currently a
+bug in snapcraft around using the 4.x channel and using remote-build, where the
+LP job created for the remote-build will attempt to use the 5.x channel instead
+of the 4.x channel. This being tracked at
+https://warthogs.atlassian.net/browse/CRAFT-568. To work-around this until the
+bug is properly fixed, you can hack the snapcraft snap by applying this patch to
+the snapcraft 4.8.3 git tag: https://pastebin.ubuntu.com/p/ZvrzghB32p/ and 
+rebuilding the snapcraft snap using snapcraft itself, then installing the 
+snapcraft snap that was built. This will force all remote-builds to use 4.x, so
+obviously the patch is not suitable for general consumption but is a temporary
+work-around until the bug is fixed properly in snapcraft upstream.
+
+```
+git clone -b 4.8.3 --single-branch --depth 1 https://github.com/snapcore/snapcraft.git
+cd snapcraft
+wget --quiet https://gist.githubusercontent.com/anonymouse64/8fc6e81dac06ed033636132b4d9215f9/raw/ea3128904d419de071d035f0c6b15b74ccfac4fa/snapcraft.patch
+git apply --ignore-whitespace snapcraft.patch
+snapcraft
+snap install snapcraft_*.snap
+```
+
+Now you can use remote-build with snapcraft on the snapd tree:
+
+```
+snapcraft remote-build --build-on=armhf
+```
+
+
+#### Splicing the snapd snap into the core snap
+
+Sometimes while developing you may need to build a version of the _core_ snap
+with a custom snapd version. The snapcraft.yaml for the core snap currently is
+complex in that it assumes it is built inside Launchpad with the 
+`snappy-dev/image` PPA enabled, so it is difficult to inject a custom version of
+snapd into this by rebuilding the core snap directly, so an easier way is to 
+actually first build the snapd snap and inject the binaries from the snapd snap
+into the core snap. This currently works since both the snapd snap and the core 
+snap have the same build base of Ubuntu 16.04, so at some point in time this 
+trick will stop working when the snapd snap starts using a build base other than
+Ubuntu 16.04, but until then, you can use the following trick to more easily get
+a custom version of snapd inside a core snap.
```

Comment:
There's some redundancy in this statement. Maybe just write this part at the end of the previous period:

"(this currently works since both the snapd snap and the core snap have the same build base of Ubuntu 16.04, so at some point in time this trick will stop working when either snap starts using a different build base)"

Reply:
done in https://github.com/snapcore/snapd/pull/11124